src/Hosting/Hosting/src/StaticWebAssets/StaticWebAssetsFileProvider.cs

```diff
@@ -111,7 +112,7 @@ private class StaticWebAssetsDirectoryRoot : IDirectoryContents
             public StaticWebAssetsDirectoryRoot(PathString remainingPath)
             {
                 // We MUST use the Value property here because it is unescaped.
-                _nextSegment = remainingPath.Value.Split("/", StringSplitOptions.RemoveEmptyEntries).FirstOrDefault();
+                _nextSegment = remainingPath.Value?.Split("/", StringSplitOptions.RemoveEmptyEntries).FirstOrDefault() ?? string.Empty; // TODO - Is this right?
```

Comment:
I think @Tratcher's recommendation is to treat any `PathString.Value` that originates from a HttpRequest instance as being non-null.